diff --git a/lib/edit_workout_screen.dart b/lib/edit_workout_screen.dart
index 9289255..a546d12 100644
--- a/lib/edit_workout_screen.dart
+++ b/lib/edit_workout_screen.dart
@@ -329,15 +329,19 @@
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
+        centerTitle: true,
+        backgroundColor: Theme.of(context).colorScheme.surfaceContainer,
         title: Text(widget.workout.title),
         actions: [
           TextButton(
             onPressed: _showSetCountDialog,
             child: Text('Сеты: $_setCount'),
           ),
           IconButton(
             icon: const Icon(Icons.save),
             tooltip: 'Сохранить',
             onPressed: () => Navigator.of(context).pop(),
           ),
+          const SizedBox(width: 8),
         ],
       ),
       body: _buildBody(),